### PR TITLE
Remove olx typedefs ol/source

### PR DESCRIPTION
**Pull request by fgravin**
Move all externs for `ol/source` (see #7947).
Finish https://github.com/openlayers/openlayers/pull/8032

### CODE REVIEW

File: src/ol/source/CartoDB.js [hunk 1]
```diff
@@ -6,13 +6,38 @@ import {assign} from '../obj.js';
...
 
+/**
+ * @typedef {Object} Options
+ * @property {ol.AttributionLike} [attributions] Attributions.
+ * @property {number} [cacheSize=2048] Cache size.
+ * @property {null|string} [crossOrigin] The `crossOrigin` attribute for loaded images.  Note that
+ * you must provide a `crossOrigin` value if you are using the WebGL renderer or if you want to
+ * access pixel data with the Canvas renderer.  See
+ * {@link https://developer.mozilla.org/en-US/docs/Web/HTML/CORS_enabled_image} for more detail.
+ * @property {ol.ProjectionLike} [projection='EPSG:3857'] Projection.
```
Comment (fredj) [inline src/ol/source/CartoDB.js]: please use `module:ol/proj~ProjectionLike` instead

File: src/ol/source/Tile.js [hunk 2]
```diff
@@ -21,7 +36,7 @@ import {wrapX, getForProjection as getTileGridForProjection} from '../tilegrid.j
  * @constructor
  * @abstract
  * @extends {ol.source.Source}
- * @param {ol.SourceTileOptions} options Tile source options.
+ * @param {module:ol/source/SourceTile~Options=} options SourceTile source options.
```
Comment (fredj) [inline src/ol/source/Tile.js]: `module:ol/source/Source/Tile~Options`

File: src/ol/source/ImageWMS.js [hunk 3]
```diff
@@ -17,14 +17,40 @@ import WMSServerType from '../source/WMSServerType.js';
...
+/**
+ * @typedef {Object} Options
+ * @property {ol.AttributionLike} [attributions] Attributions.
+ * @property {null|string} [crossOrigin] The `crossOrigin` attribute for loaded images.  Note that
+ * you must provide a `crossOrigin` value if you are using the WebGL renderer or if you want to
+ * access pixel data with the Canvas renderer.  See
+ * {@link https://developer.mozilla.org/en-US/docs/Web/HTML/CORS_enabled_image} for more detail.
+ * @property {boolean} [hidpi=true] Use the `ol.Map#pixelRatio` value when requesting
+ * the image from the remote server.
+ * @property {ol.source.WMSServerType|string} [serverType=undefined] The type of
```
Comment (fredj) [inline src/ol/source/ImageWMS.js]: `=undefined` can be removed

File: src/ol/source/TileWMS.js [hunk 4]
```diff
@@ -17,13 +17,69 @@ import {hash as tileCoordHash} from '../tilecoord.js';
...
+ * @property {function(new: ol.ImageTile, ol.TileCoord,
+ *                 ol.TileState, string, ?string,
+ *                 ol.TileLoadFunctionType)} [tileClass] Class used to instantiate image tiles.
+ * Default is {@link ol.ImageTile}.
+ * @property {ol.tilegrid.TileGrid} [tileGrid] Tile grid. Base this on the resolutions,
+ * tilesize and extent supported by the server.
+ * If this is not defined, a default grid will be used: if there is a projection
+ * extent, the grid will be based on that; if not, a grid based on a global
+ * extent with origin at 0,0 will be used..
+ * @property {ol.source.WMSServerType|string} [serverType=undefined]
```
Comment (fredj) [inline src/ol/source/TileWMS.js]: `=undefined` can be removed

File: src/ol/source/Tile.js (hunk 2 above)
Comment (ahocevar) [reply src/ol/source/Tile.js]: Actually `module:ol/source/Tile~Options`.

File: src/ol/source/ImageCanvas.js [hunk 5]
```diff
@@ -6,13 +6,34 @@ import ImageCanvas from '../ImageCanvas.js';
...
+ * @property {ol.AttributionLike} [attributions] Attributions.
+ * @property {ol.CanvasFunctionType} [canvasFunction] Canvas function.
+ * The function returning the canvas element used by the source
+ * as an image. The arguments passed to the function are: `{ol.Extent}` the
+ * image extent, `{number}` the image resolution, `{number}` the device pixel
+ * ratio, `{ol.Size}` the image size, and `{module:ol/proj/Projection~Projection}` the image
+ * projection. The canvas returned by this function is cached by the source. If
+ * the value returned by the function is later changed then
+ * `dispatchChangeEvent` should be called on the source for the source to
+ * invalidate the current cached image.
```
Comment (fredj) [inline src/ol/source/ImageCanvas.js]: Can you also backport the changes from #8026 ?

Comment (fgravin) [reply src/ol/source/ImageCanvas.js]: Done and rebased, please review. Thanks

**Pull request merged**